### ISSUE TITLE
Radio button example is missing a group label

### ISSUE THREAD

**Issue by mfairchild365** [angular/material2#15026]
#### What is the expected behavior?
The purpose of the group of radio buttons in the radio button example must be conveyed to users of screen reader software when a radio button in the group recieves focus.

#### What is the current behavior?
The purpose of the group of radio buttons in the radio button example is not conveyed to users of screen reader software when a radio button in the group recieves focus.

#### What are the steps to reproduce?
1. Using either of the following screen readers
   - NVDA 2018.4.1 + Firefox 64
   - VoiceOver for IOS (12.1.3) + Safari
2. Open the [table example](https://material.angular.io/components/radio/examples)
3. Navigate to one of the checkboxes in the buttons in the 'table with selection' example.
   - In NVDA you can press the tab key until you reach a radio button
   - In VoiceOver for iOS you can swipe right until you reach a radio button
4. Observe that the purpose of the group of radio buttons is not announced by the screen reader. It should be announced as something like "what is your favorite season?"

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
- Angular Material Example 7.2.2 located at https://material.angular.io/components/
- NVDA 2018.4.1 + FireFox 64
- VoiceOver for IOS (12.1.3) + Safari (12.1.3)

#### Is there anything else we should know?
It is also important to point out that the group label must be visible too. In the case of this example, the text 'Your favorite season is' could be considered the group label.

https://github.com/angular/material2/issues/7556 is a related issue but isn't focused on accessibility.
